lib/Basic/SourceLoc.cpp

```diff
@@ -408,6 +408,66 @@ SourceManager::getGeneratedSourceInfo(unsigned bufferID) const {
   return known->second;
 }
 
+ArrayRef<CharSourceRange> IfConfigRangeInfo::getRangesWithoutActiveBody(
+    SourceManager &SM, SmallVectorImpl<CharSourceRange> &scratch) const {
```

Comment:
Wouldn’t it be easier to just return `SmallVector<CharSourceRange, 2>` here? Avoids the scratch and the function has more knowledge that the vector will contain at most 2 elements.

Reply:
Sure, I don't really feel too strongly about it, although I should note that returning a `SmallVector<CharSourceRange, 2>` doesn't guarantee at most 2 elements :) I don't think it's a guarantee this function needs to make though, callers ought to be able to handle more than 2 elements.

Reply:
> The `N` here is just the number of in-place elements, I think @ahoppen's point is that in this case, `getRangesWithoutActiveBody` knows more about the likely number of elements then eg. a caller would.

Yes, exactly.